### ISSUE TITLE
The uv-lock hook updates uv.lock but does not commit it

### ISSUE THREAD

**Issue by Adirelle** [astral-sh/uv-pre-commit#26]
I am not sure what the uv-lock is supposed to do:

* If it is supposed to check that the lockfile is up to date with regard to pyproject.toml, it should be run with the `--locked` to report any issue .
* If it is supposed to actually update the lockfile on commit, it should also automatically add the modified uv.lock to the git index so it gets included in the commit.

Right now it seems to be something in between: the lockfile is updated but not commited, and so it appears as a modified file after the commit has been done.

**Comment by thisiswhereitype** [comment]
I think this i pre-commit behaviour, failed commands and changes to tracked files indicate a failed hook.

This hook stops you committing an environment differing to the one you have `uv sync`.

**Comment by Adirelle** [comment]
Without `--locked` (the current settings of the hook), `uv lock` updates the lockfile with success so it does not stop committing.

With `--locked`, it does not update the file but fails, but could be more useful.

**Comment by thisiswhereitype** [comment]
> Without --locked (the current settings of the hook), uv lock updates the lockfile with success so it does not stop committing.

The changed file stops the commit?

**Comment by watsonjj** [comment]
I agree that this is a problem with the hook. Also for the `pip-compile` hook of the uv-pre-commit. My expectation was that if the hook changes the file (uv.lock/requirements.txt), then the hook should:
- update the file
- fail
- therefore stopping the commit

This is how the ruff pre-commit behaves.

Current behaviour is the hook:
- updates the file
- passes
- allows the commit to proceed with the non-updated file

**Comment by watsonjj** [comment]
My mistake, I was running a pip-compile hook with a non-typical output path, and not updating the files argument of the hook.

If the files argument of the hook includes uv.lock (which it does by default), and the uv.lock file is staged, then the pre-commit will fail if the uv.lock file changes during the pre-commit.

**Comment by Adirelle** [comment]
This may be related, or not, but when I update the `pyproject.toml` for something that is not related to the dependencies, running `uv lock --locked` by hand fails as expected. However, as `uv.lock` has not changed, the hooks is not fired so it does not detect the error. You end up with a clean repository that still fail on `uv lock --locked`.

**Comment by zanieb** [comment]
Yeah sorry I don't think this is an issue with the hook. This is a pre-commit behavior.

**Comment by Adirelle** [comment]
After more thorough testing, I cannot reproduce the issue I had. Thanks for your time.
